numpy/core/src/umath/funcs.inc.src

```diff
@@ -360,9 +360,9 @@ nc_exp2@c@(@ctype@ *x, @ctype@ *r)
 static void
 nc_expm1@c@(@ctype@ *x, @ctype@ *r)
 {
-    @ftype@ a = npy_exp@c@(x->real);
-    r->real = a*npy_cos@c@(x->imag) - 1.0@c@;
-    r->imag = a*npy_sin@c@(x->imag);
+    @ftype@ a = npy_cos@c@(x->imag);
+    r->real = npy_expm1@c@(x->real) * a + (a - 1);
```

Comment:
I wonder if `cos(imag) - 1` is sufficiently accurate? `-2 * sin(imag/2)**2` would have better precision. `cexpm1` is also an optional c99 function, we might want to track its status long term.

Reply:
I did find [this](https://tinyurl.com/v4au7aq). Maybe not worth worrying about right now, but this function could probably use further improvement.

Reply:
Good point regarding using the trig identities to improve the numerics. Will update that in the code. Are you happy to leave the insights from the link you sent in a future PR?

![image](https://user-images.githubusercontent.com/966348/70889908-95d0a800-1fdb-11ea-9a76-9a9b3c6a332f.png)

Reply:
> Are you happy to leave the insights from the link you sent to a future PR?

Yes. I think this is already a significant improvement.